### PR TITLE
Enabled video share button on collection detail page

### LINKED ISSUE #187: Enable share button on collection detail page
part of #175 

Should behave the same as the share button on the video detail page.

### PR DESCRIPTION
**Pull request by gsidebo**
⚠️  **REBASED ON #188** ⚠️ 

#### What are the relevant tickets?
Closes #187

#### What's this PR do?
Enables the share button on video cards in the collection detail page

#### How should this be manually tested?
Click the link!

#### Where should the reviewer start?
`static/js/components/dialogs/EditVideoFormDialog.js`

#### Any background context you want to provide?
`VideoDetailUi` was emptied by the refactor to `commonUi`, so those files were removed

#### Screenshots (if appropriate)
ss 2017-09-11 at 19 01 39

### CODE REVIEW

**Comment by pdpinch** [comment]
Same comment as #233 -- can we move the buttons up, right aligned on the same line with the video title? Imagine what a whole page of these might look like.

File: static/js/containers/VideoDetailPage_test.js [hunk 1]
```diff
@@ -107,70 +98,21 @@ describe('VideoDetailPage', () => {
...
-    assert.deepEqual(store.getState().videos.data.get(expectedVideo.key), expectedVideo);
-    sinon.assert.calledWith(updateVideoStub, video.key, {
-      title: newTitle,
-      description: newDescription,
-    });
+    assert.isTrue(wrapper.find(".share").exists());
+    assert.isTrue(wrapper.find("ShareVideoDialog").exists());
   });
 
-  it('show the share dialog', async () => {
```
Comment (mbertrand) [inline static/js/containers/VideoDetailPage_test.js]: Bring back share dialog unit tests?

File: static/js/reducers/videoDetailUi_test.js [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-  it('sets the description', () => {
-    assertReducerResultState(setDescription, ui => ui.editDialog.description, '');
-  });
-
-
-  it('sets the edit dialog visibility', () => {
-    assertReducerResultState(setEditDialogVisibility, ui => ui.editDialog.visible, false);
-  });
-
-  it('sets the share dialog visibility', () => {
```
Comment (mbertrand) [inline static/js/reducers/videoDetailUi_test.js]: Move these tests to `commonUi_test.js`?

File: static/js/containers/VideoDetailPage_test.js (hunk 1 above)
Comment (gsidebo) [reply static/js/containers/VideoDetailPage_test.js]: The test for the content of the share dialog was replicated here: https://github.com/mitodl/odl-video-service/pull/224/files/bd4e940252a02b82ec680501abaf1b8c01fb97b7#diff-00c8a362f3a08ddcb7355a517c8f9d7eR42. The tests for showing and hiding a simple dialog are now
in [hoc_test.js](https://github.com/mitodl/odl-video-service/pull/224/files/bd4e940252a02b82ec680501abaf1b8c01fb97b7#diff-cd5b8f4619ea92fc1bf81505be1a1438).

File: static/js/reducers/videoDetailUi_test.js (hunk 2 above)
Comment (gsidebo) [reply static/js/reducers/videoDetailUi_test.js]: Done. I created a general test for `showDialog`/`hideDialog`

**Pull request merged**